engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -121,8 +143,12 @@ def verify_base_product_upgrade
           return unless upgrade_product.base?
 
           activated_bases = @system.products.where(product_type: 'base')
-          activated_bases.each do |base_product|
-            return true if (base_product.identifier == upgrade_product.identifier)
+          upgrade_product_subscription = find_subscription(upgrade_product, logger, request)
```

Comment:
It would be more straight forward to check if the `base_product_subscription` includes the `upgrade_product`: `base_product_subscription.products.include?(upgrade_product)`